client/js/app/utils/RunValidations.js

```diff
@@ -1,23 +1,28 @@
 module.exports = {
 
   run: function(validatorSet, model) {
-    var errors = [];
     var keys = Object.keys(validatorSet);
+    model.isValid = true;
+    model.errors = [];
 
     for (var i=0; i<keys.length; i++) {
       var validator = validatorSet[keys[i]];
+
+      // Check if this validator should be run at all.
       if (validator.shouldRun && !validator.shouldRun(model)) continue;
 
       var result = validator.validate(model);
+
+      // Validator returned true. This attribute is valid. Move on.
       if (result === true) continue;
-      var errorMsg = (result === false) ? validator.msg : result;
-      errors.push({
+
+      // There is an error. Set the model as invalid and set the errors on the model.
+      model.isValid = false;
+      model.errors.push({
         attribute: keys[i],
-        msg: errorMsg
+        msg: (result === false) ? validator.msg : result
```

Comment:
Nope, it can also be a string. If so, then that string should be the validation message. This is basically a crappy hack to allow a validator to return different error messages depending on the "type" of validation failure. See the `time` validator to see how it works in action.